Sources/Kitura/RouterRequest.swift

```diff
@@ -234,42 +225,46 @@ public class RouterRequest {
 }
 
 private class Cookies {
-
-    /// Storage of parsed Cookie headers
-    fileprivate var cookies = [String: HTTPCookie]()
-
-    /// Static for Cookie header key value
-    private let cookieHeader = "cookie"
-
-    fileprivate init(headers: HeadersContainer) {
-        guard let rawCookies = headers[cookieHeader] else {
-            return
+    private static var separator: RegularExpressionType = {
+        do {
+            // matches that do not contain semicolons and do not start with whitespaces
+            // effectively splits string by ";\\s*"
+            return try RegularExpressionType(pattern: "[^;\\s][^;]*", options: [])
+        } catch { // should never throw here, famous last words
+            Log.error("Error creating cookie separator regex: \(error)")
+            exit(1)
```

Comment:
I've addressed the other comments, but not sure if there's a better pattern/way to achieve this one?
The pattern literal we're using is pretty straightforward, so any changes to RegularExpression to break this would have to be really bad.